CDC-Data-Reconciliation-Backend/server.py

```diff
@@ -63,6 +63,30 @@ async def manual_report(state_file: UploadFile = File(None), cdc_file:  UploadFi
     return res
 
 
+@app.get("/reports/{report_id}")
+async def get_report(report_id: int):
+    """
+    Endpoint to fetch a report by its ID.
+    """
+    report = fetch_report_from_db(report_id)
+    if report is None:
+        raise HTTPException(status_code=404, detail="Report not found")
+    return report
+
+def fetch_report_from_db(report_id: int):
+    """
+    Function to fetch a report from the SQLite database.
+    """
+    try:
+        liteConn = sqlite3.connect("database.db") 
+        cur = liteConn.cursor()
+        cur.execute("SELECT * FROM Reports WHERE ID = ?", (report_id,))
+        report = cur.fetchone()
+        liteConn.close()
+        return report
+    except sqlite3.Error as e:
+        print(f"Database error: {e}")
+        return None
```

Comment:
There should be a newline after this line so that there is a newline between the 2 functions.